resource-managers/nomad/src/main/scala/org/apache/spark/deploy/nomad/NomadDryRun.scala

```diff
@@ -31,7 +31,7 @@ private[spark] object NomadDryRun extends Logging {
 
     val backendConf = conf.get("spark.submit.deployMode") match {
       case "cluster" => NomadClusterModeConf(conf, parseArguments(argStrings)).backend
-      case _ => NomadClusterManagerConf(conf, None)
+      case _ => NomadClusterManagerConf(conf, submitMode = false, None)
```

Comment:
minor - can we make this slightly more descriptive like `useSubmitToken` or `isInitialSubmission` or something like that